File: AndroidLab/ch9/src/main/java/com/example/ch9/section1/Test1_1Activity.java
package com.example.ch9.section1;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.ch9.R;
import com.example.ch9.databinding.ActivityTest11Binding;

public class Test1_1Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);

        ActivityTest11Binding binding = ActivityTest11Binding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        binding.button1.setOnClickListener(view -> {
            //Sub1Activity... 명시적...
            Intent intent = new Intent(this, Sub1Activity.class);
            startActivity(intent);
        });

        binding.button2.setOnClickListener(view -> {
            //Sub2Activity... 암시적....
            Intent intent = new Intent();
            intent.setAction("ACTION_SUB2");
            //android.intent.category.DEFAULT category 에 한해서는.. 정보를 주지 않아도 된다..
            //기본이 android.intent.category.DEFAULT category 에서 뒤지게 되어 있다..
            startActivity(intent);
        });

        binding.button3.setOnClickListener(view -> {
            //Sub3Activity를 암시적으로... category 정보 지정해서..
            Intent intent = new Intent();
            intent.setAction("ACTION_SUB3");
            intent.addCategory("com.example.ch9.CATEGORY_SUB3");
            startActivity(intent);
        });
        binding.button4.setOnClickListener(view -> {
            //Sub4Activity를 암시적으로 , data 정보 설정..
            Intent intent = new Intent("ACTION_SUB4", Uri.parse("http://www.naver.com"));
            startActivity(intent);
        });
    }
}

File: AndroidLab2/ch3/src/main/java/com/example/ch3/section1/Test1_1Activity.kt
package com.example.ch3.section1

import android.content.Intent
import android.os.Bundle
import android.provider.Telephony.Mms.Intents
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.ch3.R
import com.example.ch3.databinding.ActivityTest11Binding

class Test1_1Activity : AppCompatActivity() {

    var count = 0
    lateinit var binding: ActivityTest11Binding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        binding = ActivityTest11Binding.inflate(layoutInflater)
        setContentView(binding.root)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.goBtn.setOnClickListener {
            val intent = Intent(this, SomeActivity::class.java)
            startActivity(intent)
        }

        binding.incrementBtn.setOnClickListener {
            count++
            binding.resultView.text = "$count"
        }

        Log.d("kkang", "onCreate....")
    }

    override fun onStart() {
        super.onStart()
        Log.d("kkang", "onStart....")
    }

    override fun onResume() {
        super.onResume()
        Log.d("kkang", "onResume....")
    }

    override fun onPause() {
        super.onPause()
        Log.d("kkang", "onPause....")
    }

    override fun onStop() {
        super.onStop()
        Log.d("kkang", "onStop....")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("kkang", "onDestory....")
    }

    //onDestory 직전에 호출.. 액티비티 상태 데이터 저장..
    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt("count", count)
    }

    //activity 의 상태 데이터가 있는 경우만 호출..
    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        count = savedInstanceState.getInt("count")
        binding.resultView.text = "$count"
    }
}
//최초 실행.....
//onCreate....
//onStart....
//onResume....

//화면 전환.. 화면 이동...
//onPause....
//onStop....

//back 으로 되돌아 오면...
//onStart....
//onResume....

//화면 회전...
//onPause....
//onStop....
//onDestory....
//onCreate....
//onStart....
//onResume....